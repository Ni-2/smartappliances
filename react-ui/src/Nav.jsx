import React from 'react';

export default function Nav(props) {
  const { tabName, goToCards, hasLink } = props;
  return (
    <nav className="breadcrumb-main" aria-label="breadcrumb">
      <ol className="breadcrumb">
        {hasLink && (
          <li className="breadcrumb-item">
            <button type="button" className="btn btn-link my-btn-home" onClick={() => goToCards()}>
              My Smart Appliances
            </button>
          </li>
        )}
        <li className="breadcrumb-item active" aria-current="page">
          {tabName}
        </li>
      </ol>
    </nav>
  );
}
