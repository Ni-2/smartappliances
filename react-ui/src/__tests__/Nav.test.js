import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Nav from '../Nav';

it('renders Nav without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Nav />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders navigation without link', () => {
  const component = renderer.create(<Nav tabName="my cards" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders navigation with link', () => {
  const component = renderer.create(
    <Nav tabName="my device" goToCards={() => 'Go to cards function'} hasLink />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
