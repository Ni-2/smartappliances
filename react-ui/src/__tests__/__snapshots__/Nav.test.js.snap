// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navigation with link 1`] = `
<nav
  aria-label="breadcrumb"
  className="breadcrumb-main"
>
  <ol
    className="breadcrumb"
  >
    <li
      className="breadcrumb-item"
    >
      <button
        className="btn btn-link my-btn-home"
        onClick={[Function]}
        type="button"
      >
        My Smart Appliances
      </button>
    </li>
    <li
      aria-current="page"
      className="breadcrumb-item active"
    >
      my device
    </li>
  </ol>
</nav>
`;

exports[`renders navigation without link 1`] = `
<nav
  aria-label="breadcrumb"
  className="breadcrumb-main"
>
  <ol
    className="breadcrumb"
  >
    <li
      aria-current="page"
      className="breadcrumb-item active"
    >
      my cards
    </li>
  </ol>
</nav>
`;
